### PR TITLE
Replaces `kubectl get` in bootstrap

### PR DESCRIPTION
**Pull request by SomtochiAma**

### CODE REVIEW

**Comment by SomtochiAma** [comment]
@justinsb, One test is failing https://github.com/kubernetes-sigs/cluster-addons/blob/master/bootstrap/app/addon-manager_test.go#L284

File: bootstrap/app/addon-manager.go [hunk 1]
```diff
@@ -105,6 +110,21 @@ func AddonManager(env func(key string) string) (*addonManager, error) {
...
+	}
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("error building config from kubeconfig %v: %v", kubeconfigPath, err)
+	}
+	c, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	am.clientset = c
```
Comment (justinsb) [inline bootstrap/app/addon-manager.go:126]: You could try something like this...

```
func (a *AddonManager) Clientset() (kubernetes.Interface, errror) {
  if am.clientset != nil {
return am.clientset, nil
}

config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
	if err != nil {
		return nil, fmt.Errorf("error building config from kubeconfig %v: %v", kubeconfigPath, err)
	}
	c, err := kubernetes.NewForConfig(config)
	if err != nil {
		return nil, err
	}
am.clientset = c
return c, nil
}
```

File: bootstrap/app/addon-manager.go [hunk 2]
```diff
@@ -172,17 +192,16 @@ func (m *addonManager) Run() error {
 }
 
 func (m *addonManager) waitForSystemServiceAccount() {
-	var out, errOut bytes.Buffer
 	for {
-		err := m.kubectl(&out, &errOut, m.kubectlOpts, "get", "-n", systemNamespace, "serviceaccount", "default", "-o", "go-template={{with index .secrets 0}}{{.name}}{{end}}")
-		if errStr := errOut.String(); errStr != "" {
-			klog.Warning(errStr)
+		sa, err := m.clientset.CoreV1().ServiceAccounts(systemNamespace).Get(context.Background(), "default",
```
Comment (justinsb) [inline bootstrap/app/addon-manager.go:196]: And you'll have to change clientset to something like this...

```
clientset, err := Clientset()
if err != nil {
klog.Warningf("error creating kubernetes client: %v", err)
time.Sleep(500 * time.Millisecond)
continue
}
```

**Comment by justinsb** [comment]
I think this is a good change, but (as we discussed) we're going to try to get the bootstrap image under k/k e2e first, then start refactoring it.

**Pull request closed without merging**